pkg/agent/cache/category.go

```diff
@@ -0,0 +1,69 @@
+package cache
+
+import (
+	"fmt"
+
+	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+	"github.com/Axway/agent-sdk/pkg/cache"
+)
+
+// Category cache management
+
+// AddCategory - add/update Category resource in cache
+func (c *cacheManager) AddCategory(resource *v1.ResourceInstance) {
+	c.ApplyResourceReadLock()
```

Comment:
We should remove the apply and release read lock from add and delete methods, I think this was a left over mistakenly by me .